Main_project_Tests_TestUnit.rb

```diff
@@ -5,144 +5,192 @@
+		log_out
+		log_in(user)
+
+		expected_login = user.login
+		actual_login = @driver.find_element(:css, '.user.active').text
+		assert_equal(expected_login, actual_login)
+	end
+
+
+	def test_change_pass
+	 	user = register_user
+	 	change_pass(user)
+	 		
+	 	expected_notice = 'Password was successfully updated.'
+	 	actual_notice = @driver.find_element(:id, 'flash_notice').text
+	 	assert_equal(expected_notice, actual_notice)
+	 	log_out
+	 	log_in(user)
+	 	expected_login = user.login
+	 	actual_login = @driver.find_element(:css, '.user.active').text
+	 	assert_equal(expected_login, actual_login)
+	end
+
+
+ 	def test_create_new_project
+ 		register_user
+ 	 	create_new_project
+
+ 	 	expected_notice = 'Successful creation.'
+ 	 	actual_notice = @driver.find_element(:id, 'flash_notice').text
+ 	 	assert_equal(expected_notice, actual_notice)
+ 	end
 
-    # def test_open_issues_tab
-    # 	# open issues tab
-    # end
 
-    # def test_add_self_to_watchers
-    # open issues tab
-	# if there is a bug present then add self to watchers
-	# if no then create a new bug issue and then add self to watchers
-	# Verify that there is a bug issue and that current user is a watcher
-    end
+ 	def test_create_new_version
+ 		register_user
+ 		project = create_new_project
+ 		create_new_version(project)
+
+ 		expected_notice = 'Successful creation.'
+ 		actual_notice = @driver.find_element(:id, 'flash_notice').text
+ 		assert_equal(expected_notice, actual_notice)
+ 	end
 
 
+	def test_create_bug
+		register_user
+		create_new_project
+		issue = create_new_issue(type: '1')
+	     	
+		actual_array_subjects = @driver.find_elements(:class, 'subject').map(&:text)
+		assert_true(actual_array_subjects.include? issue.name)	
+	end
+
+
+	def test_create_feature
+		register_user
+		create_new_project
+		issue = create_new_issue(type: '2')
+		     	
+		expected_subject = issue.name
+		actual_array_subjects = @driver.find_elements(:class, 'subject').map(&:text)
+		assert_true(actual_array_subjects.include? expected_subject)
+	end
+
+
+	def test_create_support
+		register_user
+		create_new_project
+		issue = create_new_issue(type: '3')
+		    
+		expected_subject = issue.name
+		actual_array_subjects = @driver.find_elements(:class, 'subject').map(&:text)
+		assert_true(actual_array_subjects.include? expected_subject)
+	end
+
+
+	def test_create_issues
+		register_user
+		create_new_project
+		issue1 = create_new_issue(type: '1')
+		issue2 = create_new_issue(type: '2')
+		issue3 = create_new_issue(type: '3')
+		     	
+		actual_array_subjects = @driver.find_elements(:class, 'subject').map(&:text)
+		assert_equal(issue3.name, actual_array_subjects[0])
+		assert_equal(issue2.name, actual_array_subjects[1])
+		assert_equal(issue1.name, actual_array_subjects[2])
+	end
 
 
+    def test_open_project
+    	register_user
+    	project = create_new_project
+    	open_project(project) || tries = 3
+
+    	begin
+    		@driver.find_element(:css, '.overview.selected').displayed?
```

Comment:
define method which will raise an Exception in specific case
then you'll be able to rescue it